posthog/api/plugin.py

```diff
@@ -203,15 +208,55 @@ def check_for_updates(self, request: request.Request, **kwargs):
 
         return Response({"plugin": PluginSerializer(plugin).data})
 
+    @action(methods=["GET"], detail=True)
+    def source(self, request: request.Request, **kwargs):
+        plugin = self.get_plugin_with_permissions(reason="source editing")
+        response: Dict[str, str] = {}
+        for source in PluginSourceFile.objects.filter(plugin=plugin):
+            response[source.filename] = source.source
+        return Response(response)
+
+    @action(methods=["PATCH"], detail=True)
+    def update_source(self, request: request.Request, **kwargs):
+        plugin = self.get_plugin_with_permissions(reason="source editing")
+        sources: Dict[str, PluginSourceFile] = {}
+        performed_changes = False
+        for source in PluginSourceFile.objects.filter(plugin=plugin):
+            sources[source.filename] = source
+        for key, value in request.data.items():
+            if not sources.get(key):
+                sources[key] = PluginSourceFile.objects.create(plugin=plugin, filename=key, source=value)
+                continue
+            if sources[key].source != value:
+                performed_changes = True
+                if value is None:
+                    sources[key].delete()
+                    del sources[key]
+                else:
+                    sources[key].source = value
+                    sources[key].save()
+        response: Dict[str, str] = {}
+        for key, source in sources.items():
+            response[source.filename] = source.source
+
+        # Update values from plugin.json
+        plugin_json = json.loads(response.get("plugin.json") or "{}")
```

Comment:
```suggestion
        plugin_json = json.loads(response.get("plugin.json", "{}"))
```

Relying on behavior of falsy values is generally an antipattern in python. `.get()` already has a default value argument.

Reply:
This was explicit. The `plugin.json` string may contain `""`, and `response.get("plugin.json")` will then return `""`, while I want to get `None` instead

Reply:
That's the kind of thing sure to will lead to bugs as things get refactored, especially when not covered by _separate_ tests. Could we make the intent for "" vs None behavior more explicit via a model property or something?